dataset/src/main/scala/frameless/FramelessSyntax.scala

```diff
@@ -0,0 +1,13 @@
+package frameless
+
+import org.apache.spark.sql.{Column, Dataset}
+
+trait FramelessSyntax {
+  implicit class ColumnSyntax(self: Column) {
+    def asTyped[T, U: TypedEncoder]: TypedColumn[T, U] = new TypedColumn[T, U](self)
+  }
+
+  implicit class DatasetSyntax[T: TypedEncoder](self: Dataset[T]) {
+    def asTyped: TypedDataset[T] = new TypedDataset[T](self)
```

Comment:
I think asking for the `implicit TypedEncoder` in the def instead of in the class leads to nicer errors. (Cannot find implicit TypedEncoder instead of method asTyped not a member of Dataset)